### PR TITLE
doc: Add switch on enum example to developer notes

### PR DESCRIPTION
**Pull request by hebasto**
This PR documents a recurring issue:
- #15938
- #17105

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -563,6 +562,34 @@ class A
...
+    case Tabs::INFO: return 0;
+    case Tabs::CONSOLE: return 1;
+    case Tabs::GRAPH: return 2;
+    case Tabs::PEERS: return 3;
+    } // no default case, so the compiler can warn about missing cases
+    assert(false);
+}
+```
+
+*Rationale*: The comment documents skipping `default:` label, and it complies with `clang-format` rules. The assertion prevents firing of `-Wreturn-type` warning on some compilers.
```
Comment (MarcoFalke) [inline doc/developer-notes.md]: The warning is not wrong, though. While it is not undefined behavior to cast any `int` to an `enum class` type and pass it around (where it eventually will hit the `assert`), it seems the right thing to do for sanity.

**Comment by MarcoFalke** [comment]
ACK

**Review by promag — COMMENTED** [review]
ACK c8961c7d9fed07190628cf01f9dfad971a942b99, no excuse now, thanks!

**Comment by laanwj** [comment]
Seems like good advice to me. ACK c8961c7d9fed07190628cf01f9dfad971a942b99

**Comment by practicalswift** [comment]
ACK c8961c7d9fed07190628cf01f9dfad971a942b99

**Pull request merged**